app/src/main/java/org/mifos/selfserviceapp/presenters/UserDetailsPresenter.java

```diff
@@ -0,0 +1,125 @@
+package org.mifos.selfserviceapp.presenters;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.util.Base64;
+import android.util.Log;
+
+import org.mifos.selfserviceapp.R;
+import org.mifos.selfserviceapp.api.DataManager;
+import org.mifos.selfserviceapp.injection.ApplicationContext;
+import org.mifos.selfserviceapp.models.client.Client;
+import org.mifos.selfserviceapp.presenters.base.BasePresenter;
+import org.mifos.selfserviceapp.ui.views.UserDetailsView;
+
+import java.io.IOException;
+
+import javax.inject.Inject;
+
+import okhttp3.ResponseBody;
+import rx.Subscriber;
+import rx.android.schedulers.AndroidSchedulers;
+import rx.schedulers.Schedulers;
+import rx.subscriptions.CompositeSubscription;
+
+/**
+ * Created by naman on 07/04/17.
+ */
+
+public class UserDetailsPresenter extends BasePresenter<UserDetailsView> {
+
+    private final DataManager dataManager;
+    private CompositeSubscription subscriptions;
+
+    @Inject
+    public UserDetailsPresenter(@ApplicationContext Context context, DataManager dataManager) {
+        super(context);
+        this.dataManager = dataManager;
+        subscriptions = new CompositeSubscription();
+    }
+
+    @Override
+    public void attachView(UserDetailsView mvpView) {
+        super.attachView(mvpView);
+    }
+
+    @Override
+    public void detachView() {
+        super.detachView();
+        subscriptions.unsubscribe();
+    }
+
+    public void getUserDetails() {
+        checkViewAttached();
+        subscriptions.add(dataManager.getCurrentClient()
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribeOn(Schedulers.io())
+                .subscribe(new Subscriber<Client>() {
+                    @Override
+                    public void onCompleted() {
+
+                    }
+
+                    @Override
+                    public void onError(Throwable e) {
+                        getMvpView().showError(context.getString(R.string.error_fetching_client));
+                        getMvpView().hideProgress();
+                    }
+
+                    @Override
+                    public void onNext(Client client) {
+                        if (client != null) {
+                            getMvpView().showUserDetails(client);
+                        } else {
+                            getMvpView().showError(context
+                                    .getString(R.string.error_client_not_found));
+                        }
+                    }
+                })
+        );
+    }
+
+    public void getUserImage() {
+        checkViewAttached();
+        subscriptions.add(dataManager.getClientImage()
+                .observeOn(Schedulers.newThread())
+                .subscribeOn(Schedulers.io())
```

Comment:
@naman14 observeOn will be on `AndroidSchedulers.mainThread()` and about the async loading image you can schedule that on newThread by subscribing to new thread `Schedulers.newThread()`

Note checkout this: https://github.com/ReactiveX/RxJava/wiki/The-RxJava-Android-Module#observing-on-the-ui-thread